practice/3_python_testing/test_read_write_2.py

```diff
@@ -4,3 +4,24 @@
 https://docs.python.org/3/library/tempfile.html
 https://docs.pytest.org/en/6.2.x/tmpdir.html
 """
+import pytest
+import tempfile
+import os
+from python_part_2 import task_read_write_2 as task
+
+class TestReadWrite2:
+    """Class representing TestReadWrite2."""
+    def test_read_write_2_reverse(self):
+        """Testing that the contents of reversed file are in opposite order to contents of original file."""
+        temp_rand_file = tempfile.mkstemp()[1]
+        temp_reverse_file = tempfile.mkstemp()[1]
+        task.task_read_write_2(temp_rand_file, temp_reverse_file)
+        with open(temp_rand_file, "r", encoding="utf-8") as temp_file:
+            contents = temp_file.readlines()
+        
+        with open(temp_reverse_file, "r", encoding="utf-8") as reversed_temp_file:
+            reversed_contents = reversed_temp_file.readlines()
+        assert reversed_contents == list(reversed(contents))
+
+        os.remove(temp_rand_file)
```

Comment:
if the test fails before we execute this line, the temp file won't be deleted at all. 
Unlike [TemporaryFile()](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile), the user of [mkstemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp) is responsible for deleting the temporary file when done with it.

Is there a reason not to use TemporaryFile?
https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile

"The file is created securely, using the same rules as [mkstemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp). It will be destroyed as soon as it is closed (including an implicit close when the object is garbage collected)."